src/gc/gc.cpp

```diff
@@ -5291,45 +5334,59 @@ void set_thread_affinity_mask_for_heap(int heap_number, GCThreadAffinity* affini
     affinity->Group = GCThreadAffinity::None;
     affinity->Processor = GCThreadAffinity::None;
 
-    uintptr_t pmask, smask;
-    if (GCToOSInterface::GetCurrentProcessAffinityMask(&pmask, &smask))
+    uintptr_t pmask = process_mask;
+    int bit_number = 0; 
+    uint8_t proc_number = 0;
+    for (uintptr_t mask = 1; mask != 0; mask <<= 1)
     {
-        pmask &= smask;
-        int bit_number = 0; 
-        uint8_t proc_number = 0;
-        for (uintptr_t mask = 1; mask != 0; mask <<= 1)
+        if ((mask & pmask) != 0)
         {
-            if ((mask & pmask) != 0)
+            if (bit_number == heap_number)
             {
-                if (bit_number == heap_number)
+                dprintf (3, ("Using processor %d for heap %d", proc_number, heap_number));
+                affinity->Processor = proc_number;
+                heap_select::set_proc_no_for_heap(heap_number, proc_number);
+                if (GCToOSInterface::CanEnableGCNumaAware())
                 {
-                    dprintf (3, ("Using processor %d for heap %d", proc_number, heap_number));
-                    affinity->Processor = proc_number;
-                    heap_select::set_proc_no_for_heap(heap_number, proc_number);
-                    if (GCToOSInterface::CanEnableGCNumaAware())
+                    uint16_t node_no = 0;
+                    PROCESSOR_NUMBER proc_no;
+                    proc_no.Group = 0;
+                    proc_no.Number = (uint8_t)proc_number;
+                    proc_no.Reserved = 0;
+                    if (GCToOSInterface::GetNumaProcessorNode(&proc_no, &node_no))
                     {
-                        uint16_t node_no = 0;
-                        PROCESSOR_NUMBER proc_no;
-                        proc_no.Group = 0;
-                        proc_no.Number = (uint8_t)proc_number;
-                        proc_no.Reserved = 0;
-                        if (GCToOSInterface::GetNumaProcessorNode(&proc_no, &node_no))
-                        {
-                            heap_select::set_numa_node_for_heap(heap_number, node_no);
-                        }
+                        heap_select::set_numa_node_for_heap(heap_number, node_no);
                     }
-                    return;
                 }
-                bit_number++;
+                return;
             }
-            proc_number++;
+            bit_number++;
         }
+        proc_number++;
     }
 }
 
 bool gc_heap::create_gc_thread ()
 {
     dprintf (3, ("Creating gc thread\n"));
+
+    GCThreadAffinity affinity;
+    affinity.Group = GCThreadAffinity::None;
+    affinity.Processor = GCThreadAffinity::None;
+
+#if !defined(FEATURE_REDHAWK) && !defined(FEATURE_CORECLR)
```

Comment:
This should not be undef ifdef.

Also, note that there is same piece of core in `gc_thread_stub`. It does not sound right to have it in both places.

Reply:
oh.....that's totally a porting error. thanks for spotting it!